File: python-version/LC1865/find_sum_pairs.py
class FindSumPairs:
    def __init__(self, nums1: list[int], nums2: list[int]):
        self.__nums1 = nums1
        self.__nums2 = nums2
        self.__mp = dict()
        for num in nums2:
            self.__mp[num] = self.__mp.get(num, 0) + 1

    def add(self, index: int, val: int) -> None:
        old = self.__nums2[index]
        self.__mp[old] = self.__mp.get(old, 0) - 1
        self.__nums2[index] += val
        self.__mp[old + val] = self.__mp.get(old + val, 0) + 1

    def count(self, tot: int) -> int:
        ans = 0
        for num in self.__nums1:
            ans += self.__mp.get(tot - num, 0)
        return ans

File: go-version/leetcode/LC1895/find_sum_pairs.go
package LC1895

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/7/6 上午1:02
 * @FilePath: find_sum_pairs
 * @Description: 使用hash实现该能力
 */

type FindSumPairs struct {
	nums1 []int
	nums2 []int
	mp    map[int]int
}

func Constructor(nums1 []int, nums2 []int) FindSumPairs {
	mp := make(map[int]int)
	for _, num := range nums2 {
		mp[num]++
	}
	return FindSumPairs{nums1: nums1, nums2: nums2, mp: mp}
}

func (this *FindSumPairs) Add(index int, val int) {
	old := this.nums2[index]
	this.mp[old]--
	this.mp[old+val]++
	this.nums2[index] += val
}

func (this *FindSumPairs) Count(tot int) int {
	var ans int
	for _, num := range this.nums1 {
		if t, ok := this.mp[tot-num]; ok {
			ans += t
		}
	}
	return ans
}
